src/System.Management.Automation/engine/remoting/commands/EnterPSHostProcessCommand.cs

```diff
@@ -203,16 +235,19 @@ private Runspace CreateNamedPipeRunspace(int procId, string appDomainName)
             }
             catch (RuntimeException e)
             {
-                string msgAppDomainName = (!string.IsNullOrEmpty(appDomainName)) ? appDomainName : NamedPipeUtils.DefaultAppDomainName;
+                string msgAppDomainName = connectionInfo.AppDomainName ?? NamedPipeUtils.DefaultAppDomainName;
 
                 // Unwrap inner exception for original error message, if any.
                 string errorMessage = (e.InnerException != null) ? (e.InnerException.Message ?? string.Empty) : string.Empty;
 
                 ThrowTerminatingError(
                     new ErrorRecord(
                         new RuntimeException(
-                            StringUtil.Format(RemotingErrorIdStrings.EnterPSHostProcessCannotConnectToProcess,
-                                              msgAppDomainName, procId, errorMessage),
+                            StringUtil.Format(
+                                RemotingErrorIdStrings.EnterPSHostProcessCannotConnectToProcess,
+                                msgAppDomainName,
+                                connectionInfo.ProcessId,
```

Comment:
`ProcessId` property would be `0` by default when using `-DebugPipeName`. The error message may need to be revisited.

Reply:
added an error message for debugpipename scenarios